### DISCUSSION TITLE
Failed to resolve source for payload

### DISCUSSION THREAD

**Discussion by gheradel** [wixtoolset/issues#7931] [Questions] [answered]
We're upgrading from v3 to v4. We use a custom ba to do login and provisioning before we install the application. I've finally got the boostrapper building and running but when it goes to run the installer i am met failure.

Failed to resolve source for payload: XXX.Installer.msi, package: XXX.Installer.msi, container: (null), error: 0x80070002.
Error 0x80070002: Failed to resolve source, payload: XXX.Installer.msi, package: XXX.Installer.msi, container: (null)
Error 0x80070002: Failed to acquire payload: XXX.Installer
Failed to acquire payload: XXX.Installer.msi to working path: C:\Users\XXXX\AppData\Local\Temp\{B0C02B56-85CF-497B-88F4-7C1FA7C67A85}\XXX.Installer.msi, error: 0x80070002.
Error 0x80070002: Failed to acquire payload: XXX.Installer.msi to working path: C:\Users\XXXX\AppData\Local\Temp\{B0C02B56-85CF-497B-88F4-7C1FA7C67A85}\XXX.Installer.msi
Error 0x80070002: Failed cache action: cache package
Error 0x80070002: Failed while caching, aborting execution.

The MSI itself runs fine if i run it manually from the command line and pass it all the values it needs but not when run via the bootstrapper. I've seen some stuff about x86 vs x64 and i've made sure my installer, bootstrapper and bundle project are all x64

my bundle.wxs
```

<Wix 
  xmlns="http://wixtoolset.org/schemas/v4/wxs" 
  xmlns:bal="http://wixtoolset.org/schemas/v4/wxs/bal"
  xmlns:util="http://wixtoolset.org/schemas/v4/wxs/util">
  
  <Bundle Name="XXX" Version="1.0.0.0" Manufacturer="XXXX" UpgradeCode="XXXXXXXX" AboutUrl="XXXXXXXXX">

    <!-- Configuration variables that get passed from the Wizard through to the MSI-->
    <Variable Name="XXXVariable1" bal:Overridable="yes" />
    <Variable Name="XXXVariable2" bal:Overridable="yes" />
    <Variable Name="XXXVariable3" bal:Overridable="yes" />
    <Variable Name="XXXVariable4" bal:Overridable="yes" />
    <Variable Name="XXXVariable5" bal:Overridable="yes" />

    <!-- These are the files required to run the wizard. If you -->
    <BootstrapperApplication>
      <bal:WixDotNetCoreBootstrapperApplicationHost SelfContainedDeployment="yes" />      
      <PayloadGroupRef Id="publish.XXXX.Wix.Bootstrapper" />
    </BootstrapperApplication>

    <Chain>    
      <MsiPackage SourceFile="..\XXXX.Installer\bin\x64\$(var.Configuration)\XXX.Installer.msi">
        <!-- Map the variables to the msi so they get passed in -->
        <MsiProperty Name="XXXVariable1" Value="[XXXVariable1]" />
        <MsiProperty Name="XXXVariable2" Value="[XXXVariable2]" />
        <MsiProperty Name="XXXVariable3" Value="[XXXVariable3]" />
        <MsiProperty Name="XXXVariable4" Value="[XXXVariable4]" />
        <MsiProperty Name="XXXVariable5" Value="[XXXVariable5]" />
      </MsiPackage>
    </Chain>
  </Bundle>
</Wix>

```
Any help\tips would be appreciated.

**Comment by gheradel** [comment, marked as answer]
I solved it, In the CacheAcquireResolving we were setting the event args action incorrectly

**Reply by zubair218** [reply]
Can you please share the code?

**Reply by gheradel** [reply]
We were hooking into the CacheAcquireResolving event and had code that was setting the event args Action property.

Bootstrapper.CacheAcquireResolving += CacheAcquireResolving;


private void CacheAcquireResolving( object sender, CacheAcquireResolvingEventArgs e )
{
 e.Action = <some logic to see if we needed to download> ? CacheResolveOperation.Download : CacheResolveOperation.None;
}

We ended up just removing that method completely and it resolved the issue.
